### PR TITLE
Refactor image handling a bit

### PR DESCRIPTION
**Pull request by marcelveldt**
Refactors the logic to resolve images that are not (or may be not) remotely accessible. Instead of limiting that functionality to the file providers, this is now useable for other providers too (e.g. Plex)

### CODE REVIEW

**Comment by marcelveldt** [comment]
I'll go ahead and merge this and adjust the frontend so you can give it a test and we address any issues you may find then.

File: music_assistant/server/controllers/metadata.py [hunk 1]
```diff
@@ -273,35 +272,35 @@ async def get_image_url_for_item(
...
+                if img.provider != "url" and not resolve:
                     continue
-                if img.source != "http" and resolve_local:
-                    # return imageproxy url for local filesystem items
+                if img.provider != "url" and resolve:
+                    # return imageproxy url for images that need to be resolved
                     # the original path is double encoded
                     encoded_url = urllib.parse.quote(urllib.parse.quote(img.url))
                     return f"{self.mass.webserver.base_url}/imageproxy?path={encoded_url}"
                 return img.url
```
Comment (micha91) [inline music_assistant/server/controllers/metadata.py:282]: I think this should be img.path, right?

Comment (marcelveldt) [reply music_assistant/server/controllers/metadata.py:282]: oops, yes and thats why we invented code reviews :-)

Comment (marcelveldt) [reply music_assistant/server/controllers/metadata.py:282]: fixed it on main

**Pull request merged**